Clarify purpose of GH test workflows (#8153)
microsoft/aspire: tests/Aspire.EndToEnd.Tests/README.md

# Aspire.EndToEnd.Tests

The purpose of the `Aspire.EndToEnd.Tests` project is to run end-to-end tests against pre-built NuGet packages (nupkgs). These tests validate the ability to build, run, and interact with Aspire projects in both `in-repo` and `outside-of-repo` modes, ensuring compatibility with CI pipelines and local development environments.

> On the CI the tests are run via GitHub actions defined in `tests-integration.yml`.

Reviewer: ```suggestion
For PRs the tests are run via GitHub actions defined in `tests-integration.yml`.
```
